Galaxus.Functional.Tests/Option/Async/AsyncOptionExtensions.MatchAsyncTest.cs

```diff
@@ -0,0 +1,446 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace Galaxus.Functional.Tests.Option.Async;
+
+[TestFixture]
+public class MatchAsyncTest
+{
+    [Test]
+    public async Task Option_MatchAsyncSome_Works()
```

Comment:
Right now it's a handful of tests with many many test cases inside them? The problem I see is that when we accidentally break a single method, the whole test fails and we have to take a good look at the line number and the test code to see which method is broken.

While it's a bit more "boilerplate", what about having a single test method for each case? Exception handling aside, we got two cases for each of the `MatchAsync` methods - `Some` and `None`. You can take a look at the other async extension-tests; personally, I am a fan of the structure defined there. I'd imagine something like this:

```csharp
internal class MatchAsyncTest
{
    public class VoidReturn : MatchAsyncTask
    {
        public sealed class OnSomeAndOnNoneAreAsync : VoidReturn
        {
            public async Task SelfIsSome() { ... }

            public async Task SelfIsTaskOfSome() { ... }
        }
    }
}
```

And so on.. do you get my rambling? :D

Reply:
I used `OptionTests` as the template for these tests, but maybe it does not make sense in this case because there are many more test cases here than in `OptionTests`